### PR TITLE
Reduce number of operations in `Triangle::circumscribed_circle`

### PR DESCRIPTION
**Pull request by tomcur**
This is based on the simplified form for Cartesian coordinates in https://en.wikipedia.org/w/index.php?title=Circumcircle&oldid=1247179617#Cartesian_coordinates_2

Also adds a test to check the circle radius' sign is equal to the sign of the triangle's area.

### CODE REVIEW

File: src/triangle.rs [hunk 1]
```diff
@@ -141,11 +120,17 @@ impl Triangle {
...
+        let b = self.b - self.a;
+        let c = self.c - self.a;
+        let b_len2 = b.hypot2();
+        let c_len2 = c.hypot2();
+        let d_recip = 1. / (2. * b.cross(c));
 
-        Circle::new(self.circumcenter(), (ab * bc * ac) / (4.0 * self.area()))
+        let x = (c.y * b_len2 - b.y * c_len2) * d_recip;
+        let y = (b.x * c_len2 - c.x * b_len2) * d_recip;
+        let r = b_len2.sqrt() * c_len2.sqrt() * (c - b).hypot() * d_recip;
```
Comment (raphlinus) [inline src/triangle.rs:131]: If I'm reading this right, then a further possible optimization is to to use a single sqrt around the product of the distances squared of the sides.

**Review by raphlinus — COMMENTED** [review]
Looks reasonable to me, I think this can be committed after the rebase.

File: src/triangle.rs (hunk 1 above)
Comment (tomcur) [reply src/triangle.rs:133]: Well spotted! Nice.

**Review by raphlinus — APPROVED** [review]
Sorry to let this slip between the cracks.

**Comment by Philipp-M** [comment]
(I think you need to rebase to trigger the formatting CI, possibly because something changed on main here, Github UI is not very helpful here...)

**Comment by tomcur** [comment]
> (I think you need to rebase/merge main to trigger the formatting CI, possibly because something changed on main here, Github UI is not very helpful here...)

(Yep, that triggered it. Thanks!)

**Pull request merged**